### PR TITLE
[clang-tidy] fix performance-* warnings

### PR DESCRIPTION
**Pull request by Rechi**
## Description
fix performance-* clang-tidy warnings

## Types of change
- [x] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:
- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [x] All new and existing tests passed

### CODE REVIEW

**Review by fuzzard — APPROVED** [review]

**Review by garbear — APPROVED** [review]

File: xbmc/pvr/channels/PVRChannelGroupInternal.cpp [hunk 1]
```diff
@@ -111,7 +112,7 @@ std::vector<std::shared_ptr<CPVRChannelGroupMember>> CPVRChannelGroupInternal::
     RemoveDeletedGroupMembers(
         const std::vector<std::shared_ptr<CPVRChannelGroupMember>>& groupMembers)
 {
-  const std::vector<std::shared_ptr<CPVRChannelGroupMember>> removedMembers =
+  std::vector<std::shared_ptr<CPVRChannelGroupMember>> removedMembers =
```
Comment (ksooo) [inline xbmc/pvr/channels/PVRChannelGroupInternal.cpp:115]: Not saying this is wrong, but could you explain why this change is needed?

Comment (fuzzard) [reply xbmc/pvr/channels/PVRChannelGroupInternal.cpp:115]: My take on it is removeMembers is returned to the caller, and thats non const. So either the return type goes const, or why bother keeping this one const if it has to be cast away by the compiler anyway

Comment (ksooo) [reply xbmc/pvr/channels/PVRChannelGroupInternal.cpp:115]: Thanks, that's what I thought.

Comment (Rechi) [reply xbmc/pvr/channels/PVRChannelGroupInternal.cpp:115]: If this variable is const, it cannot be moved at the return statement.

see https://clang.llvm.org/extra/clang-tidy/checks/performance-no-automatic-move.html

**Review by ksooo — APPROVED** [review]
